### PR TITLE
add batch rpc

### PR DESCRIPTION
**Pull request by oo7ww**
* add function **search(dapp)** to get all info including isAppliable and error message
* update readme

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-sponsor-faucet/9)

### CODE REVIEW

File: faucet.js [hunk 1]
```diff
@@ -8,6 +8,21 @@ const faucetContract = require("./build/contracts/SponsorFaucet.json");
 const gas_estimation_ratio_withdraw = 1.8;
 const gas_estimation_ratio_default = 1.3;
 
+/**
+ * @dev parse encoded data
+ * @param data
+ */
+function _parse(data) {
```
Comment (boqiu) [inline faucet.js:15]: What to parse? Strongly recommend to use a meaningful function name.

File: faucet.js [hunk 2]
```diff
@@ -72,24 +88,39 @@ class Faucet {
...
+      res = await this.provider.batch([
+        { method: "cfx_call", params: [{ to: this.address, data: data }] },
+        { method: "cfx_getSponsorInfo", params: [dapp] },
+        { method: "cfx_getCollateralForStorage", params: [dapp] },
+      ]);
+
+      faucetParams = res[0];
+      sponsorInfo = res[1];
+      collateralForStorage = new BigNumber(res[2]);
+      let start = 2 + 3 * 64;
```
Comment (boqiu) [inline faucet.js:107]: any API to decode return value?

**Review by boqiu — CHANGES_REQUESTED** [review]

File: faucet.js (hunk 1 above)
Comment (oo7ww) [reply faucet.js]: modified

File: faucet.js (hunk 2 above)
Comment (oo7ww) [reply faucet.js]: use **_parseBytecode** instead

**Review by boqiu — APPROVED** [review]

**Pull request merged**